src/main/java/org/marketdesignresearch/mechlib/outcomerules/ccg/paymentrules/PaymentNorm.java

```diff
@@ -26,10 +28,21 @@ protected void setProposedValues(IMIPResult result, IMIP mip) {
     }
 
     protected IMIPResult solveProgram(IMIP program) {
-    	getMipInstrumentation().preMIP(MipInstrumentation.MipPurpose.PAYMENT, program);
-        IMIPResult result = CPLEXUtils.SOLVER.solve(program);
-        getMipInstrumentation().postMIP(MipInstrumentation.MipPurpose.PAYMENT, program, result);
-        return result;
+    	int currentSolver = 1;
+    	while(true) {
+    		try {
+    			getMipInstrumentation().preMIP(MipInstrumentation.MipPurpose.PAYMENT, program);
+    			IMIPResult result = CPLEXUtils.SOLVER.solve(program);
+    			getMipInstrumentation().postMIP(MipInstrumentation.MipPurpose.PAYMENT, program, result);
+    			return result; 
+    		} catch(MIPException ex) {
+    			// try different CPLEX solver
+    			if(currentSolver > 4)
+    				throw ex;
+    			program.setSolveParam(SolveParam.LP_OPTIMIZATION_ALG, currentSolver);
+    			currentSolver++;
+    		}
+    	} 
```

Comment:
In some cases the default solver is not able to solve the problem. Therefore try other solvers before throwing an Exception. This is the problem that occurs sometimes also with the StandardCCGConstraintGenerator generator.

Reply:
I see. Is that due to CPLEX issues only? In that case, I really like the solution. If it's a workaround that could be solved with a better LP/MIP, there's a warning light popping up in my head :)

Reply:
Yes, when running many MIP's in sequence (like in the CCG computation) it is very likely that nummerical issues arise that have nothing to do with true infeasibility. When the MIP really is infeasible it should not be solvable by any solver.

Reply:
I don't know the other LP_OPTIMIZATION_ALGs, but if you think this is a reasonable way to tackle these numerical issues, then I'm fine with it. But wouldn't the CPLEXUtils.SOLVER#solve be the right place to put this piece of code?